vapi/simulator/simulator.go

```diff
@@ -329,6 +333,16 @@ func OK(w http.ResponseWriter, val ...interface{}) {
 	StatusOK(w, s)
 }
 
+// OKWithNoContent responds with http.StatusNoContent.
+func OKWithNoContent(w http.ResponseWriter) {
```

Comment:
Thanks @pradeep288 , lgtm overall. Just a nit here, these are two functions that do the same thing and I think we can just use the `Status` func instead of either: 
```diff
modified   vapi/simulator/simulator.go
@@ -333,16 +333,6 @@ func OK(w http.ResponseWriter, val ...interface{}) {
 	StatusOK(w, s)
 }
 
-// OKWithNoContent responds with http.StatusNoContent.
-func OKWithNoContent(w http.ResponseWriter) {
-	OKStatusNoContent(w)
-}
-
-// OKStatusNoContent responds with http.StatusNoContent.
-func OKStatusNoContent(w http.ResponseWriter) {
-	Status(w, http.StatusNoContent)
-}
-
 // BadRequest responds with http.StatusBadRequest and json encoded vAPI error of type kind.
 // For use with "/rest" endpoints where the response is a "value" wrapped structure.
 func BadRequest(w http.ResponseWriter, kind string) {
@@ -2138,7 +2128,7 @@ func (s *handler) shutdown(w http.ResponseWriter, r *http.Request) {
 		switch r.URL.Query().Get(shutdown.Action) {
 		case shutdown.Cancel:
 			delayShutdown(0)
-			OKWithNoContent(w)
+			Status(w, http.StatusNoContent)
 		case shutdown.Reboot:
 			var spec shutdown.Spec
 			if s.decode(r, w, &spec) {
@@ -2146,7 +2136,7 @@ func (s *handler) shutdown(w http.ResponseWriter, r *http.Request) {
 				s.Config.Reason = spec.Reason
 				s.Config.Action = shutdown.Reboot
 				delayShutdown(spec.Delay)
-				OKWithNoContent(w)
+				Status(w, http.StatusNoContent)
 			}
 		case shutdown.PowerOff:
 			var spec shutdown.Spec
@@ -2155,7 +2145,7 @@ func (s *handler) shutdown(w http.ResponseWriter, r *http.Request) {
 				s.Config.Reason = spec.Reason
 				s.Config.Action = shutdown.PowerOff
 				delayShutdown(spec.Delay)
-				OKWithNoContent(w)
+				Status(w, http.StatusNoContent)
 			}
 		default:
 			http.NotFound(w, r)
```

Reply:
@dougm thanks for reviewing. The reason for creating a new function is to be consistent with the current pattern.

Currently, the pattern is

OK -> StatusOK -> Status 

to keep consistent, I added a new function and my pattern looks like

OKWithNoContent -> OKStatusNoContent -> Status


What's your opinion, shall I change as per your suggestion?

Reply:
`vapi.OK` is a helper for the `/rest` endpoint (with `Value` encoding)  and `StatusOK` was added for the 7.0 `/api` endpoint response encoding. In the case of `http.StatusNoContent` there's no response body, so adding another helper function isn't needed.  You could even just call `w.WriteHeader(http.StatusNoContent)`, instead of using `Status`.  In other words, we don't need a helper for each HTTP response code, we've only added helpers for those that response with an encoded response body.

Reply:
And looking at this closer, we don't need the recently added `Status` method that was recently refactored out of `StatusOK`: https://github.com/vmware/govmomi/pull/2722

Reply:
so suggestion is to use
`w.WriteHeader(http.StatusNotFound)` instead of `Status(w, http.StatusNoContent)`

Reply:
Yes

Reply:
Changed `OKWithNoContent(w)` to `w.WriteHeader(http.StatusNotFound)`